From bc29cebb184285ef6ccb083a2df6a627b158ef53 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 10 Jul 2024 19:01:20 +0200
Subject: [PATCH 655/954] fix(stm32mp2): restore PLL1 settings after STOP2
 modes

Restore the CA35 clock based on PLL1 configuration after STOP2 modes.
In this modes, the PLL1 is stopped and the VDDCPU is switched off; all the
settings are lost and must be restore after wake up.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 43c7cc18e1dd957e975b91db10b374e47f10fc8b fix(stm32mp2): restore PLL1 settings after STOP2 modes]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I403766c1823b86efdb46ecc801b8e4d8c711e0e0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/392691
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 07c293fe4b..d9ad4ef8f0 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -590,6 +590,8 @@ static void stm32_pwr_domain_suspend_finish(const psci_power_state_t
 	case PWRSTATE_LP_STOP2:
 	case PWRSTATE_LPLV_STOP2:
 		VERBOSE("STOP2 exit\n");
+		/* restore PLL1 configuration for CA35 */
+		stm32mp2_pll1_enable();
 		/* Restore STGEN and generic timer with current clock */
 		stm32mp_stgen_config(clk_get_rate(CK_KER_STGEN));
 
-- 
2.53.0

